python/idsse_common/idsse/common/utils.py

```diff
@@ -83,8 +84,7 @@ def exec_cmd(commands: Sequence[str], timeout: Optional[int] = None) -> Sequence
         Sequence[str]: Result of executing the commands
     """
     logger.debug('Making system call %s', commands)
-    # with Popen(commands, stdout=PIPE, stderr=PIPE) as proc:
-    #     out = proc.readlines()
+    # pylint: disable=consider-using-with
```

Comment:
Why did you decide to not use "with Popen"? In the current implementation is process ever closed?

Reply:
I guess I assumed it was commented out for a reason and simply silenced the warning. We can re-enable those lines.

Reply:
It is not clear to me the correct way to use subprocess. Maybe proc.kill() actually frees underlying resources, but if that is the case there is a few line lower that I'm not sure will always work. I think it good enough for now, but maybe we should create a task to figure out the correct to leverage subprocess.